src/modules/miniplayer-disable/index.js

```diff
@@ -0,0 +1,31 @@
+const $ = require('jquery');
+const debug = require('../../utils/debug');
+const settings = require('../../settings');
+const watcher = require('../../watcher');
+
+const MINIPLAYER_DISMISS = '[data-test-selector=persistent-player-mini-dismiss]';
+
+function dismiss(shouldDismiss = true) {
+    if (!shouldDismiss) return;
+    if ($(MINIPLAYER_DISMISS)[0]) {
+        if (!settings.get('disableMiniplayer')) return;
```

Comment:
im more surprised you're not just hiding this with css.. `display: none` should take care of the video element iirc